godot-core/src/obj/base.rs

```diff
@@ -41,6 +41,12 @@ impl<T: GodotClass> Base<T> {
         Base::from_obj(Gd::from_obj_sys_weak(base.as_gd().obj_sys()))
     }
 
+    /// # Safety
+    /// The returned Base is a weak pointer, so holding it will not keep the object alive. It must not be accessed after the object is destroyed.
+    pub(crate) unsafe fn from_gd(gd: &Gd<T>) -> Self {
+        Base::from_obj(Gd::from_obj_sys_weak(gd.obj_sys()))
+    }
```

Comment:
I'd name it `weak_from_gd` then.

Reply:
I just copied this from `fn from_base(...)`. Technically, all instances of `Base` are weak references. Would you like to rename the other functions as well, or should we keep them as they are?

Reply:
Hm, good point. In that case it's probably fine to keep the name as-is...